queasars/circuit_evaluation/circuit_evaluation.py

```diff
@@ -0,0 +1,122 @@
+
+
+class BaseCircuitEvaluator(ABC):
+    """Abstract class to allow a seamless exchange of circuit
+    evaluation methods in QUEASARS eigensolvers"""
+
+    @abstractmethod
+    def evaluate_circuits(self, circuits: list[QuantumCircuit], parameter_values: list[list[float]]) -> list[float]:
+        """Evaluates a list of parameterized QuantumCircuits for a list of parameter_vales. The circuit
+        at index i is evaluated for parameter_values list at index i. The matching result is at index i of
+        the returned list
+
+        :arg circuits: Quantum circuit to evaluate, must be parameterized
+        :type circuits: list[QuantumCircuit]
+        :arg parameter_values: Parameters for which the parameterized circuits shall be evaluated
+        :type parameter_values: list[list[float]]
+        :raises CircuitEvaluatorException: If the circuit evaluation fails for any reason
+        :return: The list of the resulting floating point number
+        :rtype: list[list[float]]"""
+
+    @property
+    def n_qubits(self) -> int:
+        """Get the size of the quantum circuits (in terms of qubits), which this CircuitEvaluator can evaluate
+
+        :return: the amount of qubits
+        :rtype: int
+        """
+        return 0
+
+
+class CircuitEvaluatorException(Exception):
+    """Class for exceptions caused during the evaluation of quantum circuits"""
+
+
+class OperatorCircuitEvaluator(BaseCircuitEvaluator):
+    """Class which evaluates quantum circuits by estimating the eigenvalue of the circuit for a given operator
+
+    :param qiskit_primitive: Qiskit primitive used for estimating the circuit's eigenvalue.
+        Must be an Estimator or a Sampler
+    :type qiskit_primitive: BaseEstimator | BaseSampler
+    :param operator: Operator for which the eigenvalue is estimated. If the qiskit_primitive is a sampler it
+        must be diagonal
+    :type operator: BaseOperator
+    """
+
+    def __init__(self, qiskit_primitive: Union[BaseEstimator, BaseSampler], operator: BaseOperator):
+        """Constructor Method"""
+        self.estimator: BaseEstimator
+        if isinstance(qiskit_primitive, BaseEstimator):
+            self.estimator = qiskit_primitive
+        self.measure: bool = False
+        if isinstance(qiskit_primitive, BaseSampler):
+            self.measure = True
+            self.estimator = _DiagonalEstimator(sampler=qiskit_primitive)
+        self.operator: BaseOperator = operator
+
+    def evaluate_circuits(self, circuits: list[QuantumCircuit], parameter_values: list[list[float]]) -> list[float]:
+        if self.measure:
+            circuits = [circuit.measure_all(inplace=False) for circuit in circuits]
+        evaluation_result: EstimatorResult = self.estimator.run(
+            circuits=circuits, observables=[self.operator] * len(circuits), parameter_values=parameter_values
+        ).result()
+        result_list: list[float] = list(real(evaluation_result.values))
+        return result_list
```

Comment:
real returns the real part of a complex number. This is here for typing reasons. In theory for a non hermitian operator the results may be complex. Since we cannot deal with complex numbers in the optimization we guarantee the type as a real number here. In practice nothing is lost here as our operators are typically hermitian. Nonetheless i added a short warning about this behaviour in the method's docstring.